platform/darwin/test/MGLStyleValueTests.swift

```diff
@@ -1,8 +1,61 @@
 import XCTest
 import Mapbox
 
-
+#if os(iOS) || os(watchOS) || os(tvOS)
+typealias MGLColor = UIColor
+#elseif os(macOS)
+typealias MGLColor = NSColor
+#endif
+    
 extension MGLStyleValueTests {
+    func assertColorsEqualWithAccuracy(_ actual: MGLColor, _ expected: MGLColor, accuracy: Float = 1/255) {
+        var actualComponents : [CGFloat] = [0, 0, 0, 0]
+        var expectedComponents : [CGFloat] = [0, 0, 0, 0]
+        actual.getRed(&(actualComponents[0]), green: &(actualComponents[1]), blue: &(actualComponents[2]), alpha: &(actualComponents[3]))
+        expected.getRed(&(expectedComponents[0]), green: &(expectedComponents[1]), blue: &(expectedComponents[2]), alpha: &(expectedComponents[3]))
+        for (ac, ec) in zip(actualComponents, expectedComponents) {
+            XCTAssertEqualWithAccuracy(Float(ac), Float(ec), accuracy: accuracy)
+        }
+    }
+    
+    func assertColorValuesEqual(_ actual: MGLStyleValue<MGLColor>, _ expected: MGLStyleValue<MGLColor>) {
+        if let actualConstant = actual as? MGLStyleConstantValue<MGLColor> {
+            XCTAssertTrue(expected is MGLStyleConstantValue<MGLColor>)
+            assertColorsEqualWithAccuracy(actualConstant.rawValue, (expected as! MGLStyleConstantValue<MGLColor>).rawValue)
+        } else if let actualFunction = actual as? MGLStyleFunction<MGLColor> {
+            XCTAssertTrue(expected is MGLStyleFunction<MGLColor>)
+            let expectedFunction = expected as! MGLStyleFunction<MGLColor>
+            XCTAssertEqual(actualFunction.interpolationBase, expectedFunction.interpolationBase)
+            XCTAssertEqual(actualFunction.interpolationMode, expectedFunction.interpolationMode)
+            // TODO: assert default values are equal
+            guard let actualStops = actualFunction.stops else {
+                XCTAssertNil(expectedFunction.stops)
+                return
+            }
+            let expectedStops = expectedFunction.stops!
+            XCTAssertEqual(actualStops.keys.count, expectedStops.keys.count)
+            let actualKeys = actualStops.keys.sorted(by: { String(describing: $0) < String(describing: $1) })
```

Comment:
Normally, you can call the operator directly instead of writing a new closure:

```swift
let actualKeys = actualStops.keys.sorted(by: <)
```

This syntax relies on type inference to choose the right operator. I’m not sure if it’s a problem that the compiler thinks the keys are of type `Any` rather than `Comparable`.

The original keys are `String`s in some cases and NSNumbers (or perhaps `Float`s?) in other cases. With NSNumber, you’d probably need `.floatValue` or at least `as Float`. With the current code, you’re going to wind up with some weird sorting in the case of NSNumber. For example, `"100" < "2"`. Fortunately, there shouldn’t be a problem as long as both arrays are sorted consistently.